src/Aspire.Hosting.Azure/Provisioning/Provisioners/AzureProvisioner.cs

```diff
@@ -272,6 +278,9 @@ await provisioner.GetOrCreateResourceAsync(
                     cancellationToken).ConfigureAwait(false);
 
                 resource.ProvisioningTaskCompletionSource?.TrySetResult();
+
+                var afterResourceCreatedEvent = new AfterResourceStartedEvent(resource, serviceProvider);
+                await eventing.PublishAsync(afterResourceCreatedEvent, cancellationToken).ConfigureAwait(false);
```

Comment:
This makes me wonder if we need another event that fires on failure since these only fire on success.

Reply:
I think that would be a Bicep specific event.